compiler/src/dotty/tools/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -106,7 +106,7 @@ trait BCodeSkelBuilder extends BCodeHelpers {
 
       initJClass(cnode)
 
-      val methodSymbols = for (f <- cd.symbol.info.decls.toList if f.is(Method) && f.isTerm && !f.is(Module)) yield f
+      val methodSymbols = for (f <- cd.symbol.info.decls.iterator if f.is(Method) && f.isTerm && !f.is(Module)) yield f
```

Comment:
Ah, I see you can't directly use `decls` because it doesn't have a withFilter. In that case, using `toList` is fine, but we definitely shouldn't use `iterator`, because now `methodSymbols` is an iterator and if we use it twice it will do the wrong thing.

Reply:
Reverted